src/rdkafka_admin.c

```diff
@@ -8384,6 +8416,329 @@ rd_kafka_DescribeConsumerGroupsResponse_parse(rd_kafka_op_t *rko_req,
+                               RD_KAFKA_RESP_ERR_GROUP_ID_NOT_FOUND ||
+                           error->code ==
+                               RD_KAFKA_RESP_ERR_UNSUPPORTED_VERSION) {
+                        grpdesc = rd_kafka_ConsumerGroupDescription_new_error(
+                            group_id, error,
+                            RD_KAFKA_CONSUMER_GROUP_TYPE_CONSUMER);
+                } else {
+                        grpdesc = rd_kafka_ConsumerGroupDescription_new_error(
+                            group_id, error,
+                            RD_KAFKA_CONSUMER_GROUP_TYPE_UNKNOWN);
+                }
+
+                rd_list_add(&rko_result->rko_u.admin_result.results, grpdesc);
+
+                rd_list_destroy(&members);
+                rd_free(group_id);
+                rd_free(group_state);
+                rd_free(assignor_name);
+                rd_free(error_str);
+                RD_IF_FREE(error, rd_kafka_error_destroy);
+                RD_IF_FREE(operations, rd_free);
+
+                error         = NULL;
+                group_id      = NULL;
+                group_state   = NULL;
+                assignor_name = NULL;
+                error_str     = NULL;
+                operations    = NULL;
+        }
+        RD_IF_FREE(host, rd_free);
+        RD_IF_FREE(node, rd_kafka_Node_destroy);
+        rd_kafka_buf_skip_tags(reply);
+        *rko_resultp = rko_result;
+        return RD_KAFKA_RESP_ERR_NO_ERROR;
+err_parse:
+        RD_IF_FREE(group_id, rd_free);
+        RD_IF_FREE(group_state, rd_free);
+        RD_IF_FREE(assignor_name, rd_free);
+        RD_IF_FREE(error_str, rd_free);
+        RD_IF_FREE(host, rd_free);
+        RD_IF_FREE(node, rd_kafka_Node_destroy);
+        RD_IF_FREE(member_id, rd_free);
+        RD_IF_FREE(instance_id, rd_free);
+        RD_IF_FREE(client_id, rd_free);
+        RD_IF_FREE(client_host, rd_free);
+        RD_IF_FREE(error, rd_kafka_error_destroy);
+        RD_IF_FREE(operations, rd_free);
+        RD_IF_FREE(assignment, rd_kafka_topic_partition_list_destroy);
+        RD_IF_FREE(target_assignment, rd_kafka_topic_partition_list_destroy);
+        RD_IF_FREE(rko_result, rd_kafka_op_destroy);
+
+        rd_snprintf(
+            errstr, errstr_size,
+            "DescribeConsumerGroups response protocol parse failure: %s",
+            rd_kafka_err2str(reply->rkbuf_err));
+        return reply->rkbuf_err;
+}
+
+/**
+ * @brief In case if we get an Unsupported Feature error or if it is a consumer
+           group and we get errors GROUP_ID_NOT_FOUND(69) or
+           UNSUPPORTED_VERSION(35) we need to send a request to the old
+           protocol.
+ */
+static rd_bool_t do_fallback_to_old_consumer_group_descibe(
+    rd_kafka_ConsumerGroupDescription_t *groupres) {
+        if ((groupres->error &&
+             groupres->error->code == RD_KAFKA_RESP_ERR__UNSUPPORTED_FEATURE) ||
+            (groupres->type == RD_KAFKA_CONSUMER_GROUP_TYPE_CONSUMER &&
```

Comment:
Check `RD_KAFKA_CONSUMER_GROUP_TYPE_CONSUMER` for  `UNSUPPORTED_FEATURE` error as well. If we get this error (which might happen in old brokers), we will keep on looping.